aptos-move/framework/aptos-framework/sources/vesting.move

```diff
@@ -243,31 +243,66 @@ module aptos_framework::vesting {
         amount: u64,
     }
 
+    #[view]
+    /// Return the address of the underlying stake pool (separate resource account) of the vesting contract.
+    ///
+    /// This errors out the vesting contract with the provided address doesn't exist.
     public fun stake_pool_address(vesting_contract_address: address): address acquires VestingContract {
         assert_vesting_contract_exists(vesting_contract_address);
         borrow_global<VestingContract>(vesting_contract_address).staking.pool_address
     }
 
+    #[view]
+    /// Return the vesting start timestamp (in seconds) of the vesting contract.
+    /// Vesting will start at this time, and once a full period has passed, the first vest will become unlocked.
+    ///
+    /// This errors out the vesting contract with the provided address doesn't exist.
     public fun vesting_start_secs(vesting_contract_address: address): u64 acquires VestingContract {
         assert_vesting_contract_exists(vesting_contract_address);
         borrow_global<VestingContract>(vesting_contract_address).vesting_schedule.start_timestamp_secs
     }
 
+    #[view]
+    /// Return the duration of one vesting period (in seconds).
+    /// Each vest is released after one full period has started, starting from the specified start_timestamp_secs.
+    ///
+    /// This errors out the vesting contract with the provided address doesn't exist.
+    public fun period_duration_secs(vesting_contract_address: address): u64 acquires VestingContract {
+        assert_vesting_contract_exists(vesting_contract_address);
+        borrow_global<VestingContract>(vesting_contract_address).vesting_schedule.period_duration
+    }
+
+    #[view]
+    /// Return the remaining grant, consisting of unvested coins that have not been distributed to shareholders.
```

Comment:
> The `shareholder` function is super handy. I wonder if we could go a step further and make the `accumulated_rewards` function call `shareholder` internally? Or provide another function that wraps the two calls. Because if this isn't provided, I'll have to do it myself, either with a "view script function" (??) or two view function calls.

Thanks for the suggestion - that makes sense. The recent I refrained from using shareholder is because it's a linear scan. But since the primary use case of accumulated_rewards is off chain, that cost is fine (no gas charged anyway). So I updated accumulated_rewards to allow passing either shareholder or beneficiary

Reply:
Added